### PR TITLE
updated with Rational{Int}

### LINKED ISSUE #56: `obj` fp format instability with some problems from `unconstrained_nlp_set()`
```julia
using SolverTest
using NLPModels

T = Float16
problems = SolverTest.unconstrained_nlp_set()
for nlp in problems
  outtype = typeof(obj(nlp,T.(nlp.meta.x0)))
  typeof(obj(nlp,T.(nlp.meta.x0))) == T || println("Problem $(nlp.meta.name) is type unstable for obj. Input is Vector{$T}, output is $outtype".)
end
```

The issue comes from `dot` being called with `A` and `D` matrices that are `Float64`.

### PR DESCRIPTION
**Pull request by Jay-sanjay**
This PR is related to issue #56 

## PR CheckList

- [x] added  Rational{Int} check for unconstrained.jl and bound-constrained.jl

Close #56

### CODE REVIEW

File: src/nlp/bound-constrained.jl [hunk 1]
```diff
@@ -2,8 +2,8 @@ export bound_constrained_nlp
 
 function bound_constrained_nlp_set(;kwargs...)
   n = 30
-  D = Diagonal([0.1 + 0.9 * (i - 1) / (n - 1) for i = 1:n])
-  A = spdiagm(0 => 2 * ones(n), -1 => -ones(n - 1), -1 => -ones(n - 1))
+  D = Diagonal([Rational{Int}(1//10 + 9//10 * (i - 1) / (n - 1)) for i = 1:n])
```
Comment (tmigot) [inline src/nlp/bound-constrained.jl:5]:
```suggestion
  D = Diagonal([1//10 + 9//10 * (i - 1) // (n - 1) for i = 1:n])
```

File: src/nlp/unconstrained.jl [hunk 2]
```diff
@@ -2,8 +2,8 @@ export unconstrained_nlp
 
 function unconstrained_nlp_set(;kwargs...)
   n = 30
-  D = Diagonal([0.1 + 0.9 * (i - 1) / (n - 1) for i = 1:n])
-  A = spdiagm(0 => 2 * ones(n), -1 => -ones(n - 1), -1 => -ones(n - 1))
+  D = Diagonal([Rational{Int}(1//10 + 9//10 * (i - 1) / (n - 1)) for i = 1:n])
```
Comment (tmigot) [inline src/nlp/unconstrained.jl:5]:
```suggestion
  D = Diagonal([1//10 + 9//10 * (i - 1) // (n - 1) for i = 1:n])
```

**Review by tmigot — CHANGES_REQUESTED** [review]
Thanks @Jay-sanjay that looks great!
Could you also do the same for these two?
- [ ] https://github.com/Jay-sanjay/SolverTest.jl/blob/a695b84058251b435435feb8dd37df356dfd4f05/src/nls/unconstrained.jl#L5
- [ ] https://github.com/Jay-sanjay/SolverTest.jl/blob/a695b84058251b435435feb8dd37df356dfd4f05/src/nls/bound-constrained.jl#L5

**Comment by Jay-sanjay** [comment]
Thank you @tmigot sir, I guess I just missed to update that nls file before asking for the review :)

**Comment by Jay-sanjay** [comment]
Thank you @tmigot sir, I guess I just missed to update that nls file before asking for the review :)
Sorry for that, I will try to recheck things before asking for the review

**Pull request merged**